### PR TITLE
Add support for Azure Linux 3

### PR DESCRIPTION
**Pull request by narrieta**

### CODE REVIEW

File: tests_e2e/test_suites/images.yml [hunk 1]
```diff
@@ -17,8 +17,8 @@ image-sets:
       - "debian_11"
       - "flatcar"
       - "suse_12"
-      - "mariner_1"
```
Comment (narrieta) [inline tests_e2e/test_suites/images.yml:20]: Mariner 1 is deprecated now

File: tests_e2e/tests/lib/agent_log.py [hunk 2]
```diff
@@ -380,6 +380,24 @@ def get_errors(self) -> List[AgentLogRecord]:
...
+            # TODO: The Daemon has not been updated on Azure Linux 3; remove this message when it is.
+            #
+            # 2024-08-05T14:36:48.004865Z WARNING Daemon Daemon Unable to load distro implementation for azurelinux. Using default distro implementation instead.
+            #
+            {
+                'message': r"Unable to load distro implementation for azurelinux. Using default distro implementation instead.",
+                'if': lambda r: DISTRO_NAME == 'azurelinux' and r.prefix == 'Daemon' and r.level == 'WARNING'
+            },
+            #
+            # TODO: The OMS extension does not support Azure Linux 3; remove this message when it does.
```
Comment (maddieford) [inline tests_e2e/tests/lib/agent_log.py:393]: Should we skip any tests which are installing OMS instead? Or is it added by policy?

Comment (narrieta) [reply tests_e2e/tests/lib/agent_log.py:393]: it's policy

**Review by maddieford — APPROVED** [review]

**Review by nagworld9 — DISMISSED** [review]

File: azurelinuxagent/common/osutil/factory.py [hunk 3]
```diff
@@ -142,7 +142,7 @@ def _get_osutil(distro_name, distro_code_name, distro_version, distro_full_name)
     if distro_name == "iosxe":
         return IosxeOSUtil()
 
-    if distro_name == "mariner":
+    if distro_name in ["mariner", "azurelinux"]:
```
Comment (nagworld9) [inline azurelinuxagent/common/osutil/factory.py:145]: I think same change need in setup.py too

**Review by maddieford — APPROVED** [review]

**Review by nagworld9 — APPROVED** [review]

**Pull request merged**